awx/ui/client/src/network-ui/network-details/details.controller.js

```diff
@@ -0,0 +1,44 @@
+/*************************************************
+ * Copyright (c) 2016 Ansible, Inc.
+ *
+ * All Rights Reserved
+ *************************************************/
+
+ export default
+ 	['$scope', '$state', '$stateParams', 'GenerateForm', 'ParseTypeChange', 'HostsService',
+ 	function($scope, $state, $stateParams, GenerateForm, ParseTypeChange, HostsService){
+
+        $scope.parseType = 'yaml';
+ 		$scope.formCancel = function(){
+            $scope.$parent.$$childTail.closeDetailsPanel();
+ 		};
+
+ 		$scope.formSave = function(){
+ 			var host = {
+ 				id: $scope.item.id,
+ 				variables: $scope.variables === '---' || $scope.variables === '{}' ? null : $scope.variables,
+ 				name: $scope.item.name,
+ 				description: $scope.item.description,
+ 				enabled: $scope.item.enabled
+ 			};
+ 			HostsService.put(host).then(function(response){
+                $scope.saveConfirmed = true;
+                if(_.has(response, "data")){
+                    $scope.$parent.$broadcast('awxNet-hostUpdateSaved', response.data);
```

Comment:
Talked with @jaredevantabor about mitigating the use of the $broadcast/$on pattern if possible.  I don't consider this to be something that would block the merge but is a nice thing to clean up _if possible_ in the future.

Reply:
Yes, I agree broadcasting messages should be avoided.   This should be changed to unicast a message to a specific listener.

Reply:
Yes, I'm planning on doing some cleanup in the near future to consolidate the scopes being used between the shell and the canvas itself. This will help mitigate the need for these $on/$broadcast/$emit events.